File: crabzilla-core/src/main/java/io/github/crabzilla/core/entity/Snapshot.java
package io.github.crabzilla.core.entity;

import lombok.Value;

@Value
public class Snapshot<A extends Entity> {

  private final A instance;
  private final Version version;

  public Version nextVersion() {
    return version.nextVersion();
  }

  public boolean isEmpty() {
    return version.getValueAsLong()==0L;
  }

}

File: crabzilla-core/src/main/java/io/github/crabzilla/core/entity/Snapshot.kt
package io.github.crabzilla.core.entity

import io.github.crabzilla.core.Version


data class Snapshot<out A : Entity>(val instance: A, val version: Version) {

  val isEmpty: Boolean
    get() = version.valueAsLong == 0L

  fun nextVersion(): Version {
    return version.nextVersion()
  }

}
